src/serialize.h

```diff
@@ -126,24 +126,28 @@ template<typename Stream> inline uint64_t ser_readdata64(Stream &s)
 }
 inline uint64_t ser_double_to_uint64(double x)
 {
+    static_assert(sizeof(double) == sizeof(uint64_t), "Floating-point width assumption");
```

Comment:
Fixed width integer types `uint32_t` and `uint64_t` are provided only if the implementation directly supports these types. So, `sizeof(uint64_t) == 8` is correct always. Could it be simpler, like this:
```
static_assert(sizeof(double) == 8, "Floating-point width assumption");
```